docs/reference.md

```diff
@@ -966,6 +983,195 @@ users from seeing authenticated pages momentarily during page load. See the
+```js
+$scope.storage.$getDownloadURL().then(function(url) {
+  $scope.url = url;
+});
+```
+
+## $getMetadata()
+
+Returns a promise of [the metadata of the file](https://firebase.google.com/docs/storage/web/file-metadata#get_file_metadata) stored at the configured path. File
+metadata contains common properties such as `name`, `size`, and `contentType`
+(often referred to as MIME type) in addition to some less common ones like `contentDisposition` and `timeCreated`.
+
+```js
+$scope.storage.$getMetadata().then(function(metadata) {
+  $scope.metadata = metadata;
+});
+```
+
+## $updateMetadata(metadata)
+
+Updates [the metadata of the file](https://firebase.google.com/docs/storage/web/file-metadata#update_file_metadata) stored at the configured path.
+Returns a promise containing the complete metadata or en error.
+
+```js
+var updateData = { contenType: 'text/plain' };
+$scope.storage.$updateMetadata(updateData).then(function(completeMetadata) {
+  $scope.completeMetadata = completeMetadata;
+});
+```
+
+## $delete()
+
+Permanently [deletes the file stored](https://firebase.google.com/docs/storage/web/delete-files) at the configured path. Returns a promise that is resolved when the delete completes.
+
+```js
+$scope.storage.$delete().then(function() {
+  console.log('successfully deleted!');
+});
+```
+
+## $toString()
+
+Returns a [string version of the bucket path](https://firebase.google.com/docs/reference/js/firebase.storage.Reference#toString) stored as a `gs://` scheme.
+
+```js
+// gs://<bucket>/<path>/<to>/<object>
+var asString = $scope.storage.$toString();
+```
+
+## Upload Task
+
+The `$firebaseStorage()` service returns an AngularFire wrapped [`UploadTask`](https://firebase.google.com/docs/reference/js/firebase.storage#uploadtask) when uploading binary content
+using the [`$put()`]($put) and [`$putString()`](#putstring) methods. This task is used for [monitoring](https://firebase.google.com/docs/storage/web/upload-files#monitor_upload_progress)
+and [managing](https://firebase.google.com/docs/storage/web/upload-files#manage_uploads) uploads.
+
+```js
+var htmlFile = new Blob(['<html></html>'], {type : 'text/html'});
+var uploadTask = $scope.storage.$put(htmlFile, { contentType: 'text/html' });
+```
+
+## $progress(callback)
+
+Calls the provided callback function whenever there is an update in the progress of the file uploading. The callback
+passes back an [`UploadTaskSnapshot`](https://firebase.google.com/docs/reference/js/firebase.storage.UploadTaskSnapshot).
+
+```js
+var uploadTask = $scope.storage.$put(file);
+uploadTask.$progress(function(snapshot) {
+  var percentUploaded = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+  console.log(percentUploaded)
+});
+```
+
+## $error(callback)
+
+Calls the provided callback function when there is an error uploading the file.
+
+```js
+var uploadTask = $scope.storage.$put(file);
+uploadTask.$error(function(error) {
+  console.error(error);
+});
+```
+
+## $complete(callback)
```

Comment:
Nit: I'd move this method before the `$error` method and update the Table of Contents ordering as well.